### ISSUE TITLE
Layered geoshape stroke properties overwrite each other

### ISSUE THREAD

**Issue by jakevdp** [vega/vega-lite#5655]
Raised in https://stackoverflow.com/questions/59222697/overlay-state-outlines-on-choropleth-in-altair/59222852#59222852

In this chart, the county strokes should be white, while the state strokes should be black; however both are black ([vega editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEA2CGCeBTATpAXGA2uCZS90gBNYAXWdPHAiSAV2WkuLNgHo6BnAWgEYAGALYA6AFacA9gDtIAGmo1IAMwnJBZSsGWIyDRMwDGEulNIBLRJzlRS8AA76MkUhLsTx0yAF8FEL-JoodWQAa01ne0coAHNECU4AC1gHa0hOUmQJEKjIAHcEs1J9f18oO0zRRANzTwwtWxSnWGgAIxROAFVOCh8CEoJ8RRJyTVLaBiYnYfYuPiExSRkAwKgVNQ067V1kHPSyS1SGnJc3Dxlemn7FYLDNo+ZY+KSU2Sh0zOzmFrgDMNflMzQaAAeTssAMhXglH4VwIkHK7iqNRkd0izGabWQnW63gUFwAuss8mYiKQEpQAOz8fhEhKIMzRBKkSgAFmpIC8QA)):
```json
{
  "layer": [
    {
      "data": {
        "url": "data/us-10m.json",
        "format": {"feature": "counties", "type": "topojson"}
      },
      "mark": {"type": "geoshape", "stroke": "white"},
      "projection": {"type": "albersUsa"}
    },
    {
      "data": {
        "url": "data/us-10m.json",
        "format": {"feature": "states", "type": "topojson"}
      },
      "mark": {"type": "geoshape", "stroke": "black", "fillOpacity": 0},
      "projection": {"type": "albersUsa"}
    }
  ],
  "width": 700,
  "height": 400
}
```
![visualization (71)](https://user-images.githubusercontent.com/781659/70368725-6bf0e600-1863-11ea-806a-ce37c1248d2f.png)

Through some experimentation, I found that if I change **one** of the dataset URLs from ``"data/us-10m.json"`` to ``"data/us-10m.json#"``, the result is the expected output:
```
{
  "layer": [
    {
      "data": {
        "url": "data/us-10m.json#",
        "format": {"feature": "counties", "type": "topojson"}
      },
      "mark": {"type": "geoshape", "stroke": "white"},
      "projection": {"type": "albersUsa"}
    },
    {
      "data": {
        "url": "data/us-10m.json",
        "format": {"feature": "states", "type": "topojson"}
      },
      "mark": {"type": "geoshape", "stroke": "black", "fillOpacity": 0},
      "projection": {"type": "albersUsa"}
    }
  ],
  "width": 700,
  "height": 400
}
```
![visualization (72)](https://user-images.githubusercontent.com/781659/70368730-92168600-1863-11ea-8fa4-4f0a3da02c22.png)

It appears that there is something about how vega handles duplicate datasets that causes stroke properties to be cross-applied.

**Comment by domoritz** [comment]
The issue here is actually that Vega-Lite merges the two datasets and so both layers use the counties.

**Comment by domoritz** [comment]
Fixed in #5656.
